inference/inference.py

```diff
@@ -199,14 +195,18 @@ def autoregressive_inference(params, ic, valid_data_full, model):
     if params.masked_acc:
       maskarray = torch.as_tensor(np.load(params.maskpath)[0:720]).to(device, dtype=torch.float)
 
-    valid_data = valid_data_full[ic:(ic+prediction_length*dt+n_history*dt):dt, in_channels, 0:720] #extract valid data from first year
+    valid_data = valid_data_full[ic:(ic+prediction_length*dt+n_history*dt):dt, in_channels] #extract valid data from first year
+    if valid_data.shape[2] > 720:
```

Comment:
Is this the longitude?

Reply:
Actually latitude. The lat dimension size in ERA5 data is 721 (goes from -90° to 90°, inclusive, at 0.25° intervals) but FourCastNet requires an image size that is an integer multiple of the patch size, and so they were just dropping the last latitude to get a dimension size of 720. This should probably be hidden within the data loader, but currently is done at semi random places throughout the code, and I was trying to avoid too much refactoring.